src/marqo/core/inference/models/hugging_face_model.py

```diff
@@ -0,0 +1,201 @@
+import os
+import tarfile
+import zipfile
+from typing import Tuple, Callable
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from pydantic import ValidationError
+from transformers import (AutoModel, AutoTokenizer)
+
+from marqo import marqo_docs
+from marqo.core.inference.model_download import download_model
+from marqo.core.inference.models.abstract_embedding_model import AbstractEmbeddingModel
+from marqo.core.inference.models.hugging_face_model_properties import HuggingFaceModelProperties, PoolingMethod
+from marqo.s2_inference.configs import ModelCache
+from marqo.s2_inference.errors import InvalidModelPropertiesError
+from marqo.s2_inference.types import Union, FloatTensor, List
+
+
+def _average_pool_func(model_output, attention_mask):
+    last_hidden = model_output.last_hidden_state.masked_fill(~attention_mask[..., None].bool(), 0.0)
+    return last_hidden.sum(dim=1) / attention_mask.sum(dim=1)[..., None]
+
+
+def _cls_pool_func(model_output, attention_mask = None):
+    return model_output[0][:,0]
+
+
+class HuggingFaceModel(AbstractEmbeddingModel):
+    """The concrete class for all sentence transformers models loaded from Hugging Face."""
+
+    def __init__(self, model_properties: dict, device: str, model_auth: dict):
+        super().__init__(model_properties, device, model_auth)
+
+        self.model = None
```

Comment:
You should prepend these with _ to mark them as private/protected

Reply:
yes, done